### DISCUSSION TITLE
Custom transients cannot take effect

### DISCUSSION THREAD

**Discussion by improve100** [magit/transient#253] [Support] [answered]
I have added two transients to my emacs configuration file, but their behavior is somewhat strange. The git svn dcommit option works fine, but the git svn rebase option doesn't.

```
(defun magit-push-to-svn ()
  (interactive)
  (magit-git-command-topdir "git svn dcommit"))
(transient-append-suffix 'magit-push "p"
  '("D" "Push to svn" magit-push-to-svn))

(defun magit-pull-from-svn ()
  (interactive)
  (magit-git-command-topdir "git svn rebase"))
(transient-append-suffix 'magit-pull "F"
  '("R" "Pull from svn" magit-pull-from-svn))
```
![image](https://github.com/magit/transient/assets/6092914/15b4ee54-9305-4299-a7ca-2d9bbedbc687)
![image](https://github.com/magit/transient/assets/6092914/56a7e9d9-b646-4b35-874d-5bb78999f071)

**Comment by tarsius** [comment, marked as answer]
You've added the command to this group:

```
  ["Fetch from"
   :if-non-nil magit-pull-or-fetch
   ("f" "remotes"           magit-fetch-all-no-prune)
   ("F" "remotes and prune" magit-fetch-all-prune)]
```

but then you apparently also `(setq magit-pull-or-fetch nil)`.

(I *think* `git svn rebase` is more like rebasing than fetching, so placing that among the fetching commands might not make much sense anyway.)

Instead try placing the command in a new group:

```elisp
(transient-append-suffix 'magit-pull '(1)
  ["Subversion"
   ("R" "Pull from svn" magit-pull-from-svn)])
```
